sqlalchemy_teradata/dialect.py

```diff
@@ -116,4 +116,154 @@ def get_view_names(self, connection, schema=None, **kw):
         res = connection.execute(stmt).fetchall()
         return [self.normalize_name(name['tablename']) for name in res]
 
+    def get_pk_constraint(self, connection, table_name, schema=None, **kw):
+        """
+        Override
+        TODO: Check if we need PRIMARY Indices or PRIMARY KEY Indices
+        TODO: Check for border cases (No PK Indices)
+        """
+
+        if schema is None:
+            schema = self.default_schema_name
+
+        stmt = select([column('ColumnName'), column('IndexName')],
+                      from_obj=[text('dbc.Indices')]).where(
+                          and_(text('DatabaseName = :schema'),
+                              text('TableName=:table'),
+                              text('IndexType=:indextype'))
+                      ).order_by(asc(column('IndexNumber')))
+
+        # K for Primary Key
+        res = connection.execute(stmt, schema=schema, table=table_name, indextype='K').fetchall()
+
+        index_columns = list()
+        index_name = None
+
+        for index_column in res:
+            index_columns.append(index_column)
+            index_name = index_column.IndexName # There should be just one IndexName
+
+        return {
+            "constrained_columns": index_columns,
+            "name": index_name
+        }
+
+    def get_unique_constraints(self, connection, table_name, schema=None, **kw):
+        """
+        Overrides base class method
+        """
+        if schema is None:
+            schema = self.default_schema_name
+
+        stmt = select([column('ColumnName'), column('IndexName')], from_obj=[text('dbc.Indices')]) \
+            .where(and_(text('DatabaseName = :schema'),
+                        text('TableName=:table'),
+                        text('IndexType=:indextype'))) \
+            .order_by(asc(column('IndexName')))
+
+        # U for Unique
+        res = connection.execute(stmt, schema=schema, table=table_name, indextype='U').fetchall()
+
+        def grouper(fk_row):
+            return {
+                'name': fk_row.IndexName,
+            }
+
+        unique_constraints = list()
```

Comment:
Yes that's the reason of those TODOs.

Glad you appreciated it! I will continue pushing PRs as long as I find something I can add/fix/improve.